### PR TITLE
Evmos transaction indexing issues

### PR DESCRIPTION
**Pull request by tzdybal**
This is experimental branch, PR  only for visibility.

### CODE REVIEW

**Comment by tzdybal** [comment]
Actually there is no reason not to merge this additional test / small improvement in the code.

File: state/executor.go [hunk 1]
```diff
@@ -303,9 +303,9 @@ func (e *BlockExecutor) execute(ctx context.Context, state State, block *types.B
 	}
 
 	for _, tx := range block.Data.Txs {
-		res := e.proxyApp.DeliverTxAsync(abci.RequestDeliverTx{Tx: tx})
-		if res.GetException() != nil {
-			return nil, errors.New(res.GetException().GetError())
+		_ = e.proxyApp.DeliverTxAsync(abci.RequestDeliverTx{Tx: tx})
+		if e.proxyApp.Error() != nil {
+			return nil, err
```
Comment (adlerjohn) [inline state/executor.go:308]: Are you sure this is right? you're returning the `err` from https://github.com/celestiaorg/optimint/blob/935c60a3e1ab2e13c34bf4cdca0dd6ce37e32e1f/state/executor.go#L291 here.

File: rpc/client/client_test.go [hunk 2]
```diff
@@ -90,8 +89,8 @@ func TestGenesisChunked(t *testing.T) {
 
 	mockApp := &mocks.Application{}
 	mockApp.On("InitChain", mock.Anything).Return(abci.ResponseInitChain{})
-	privKey, _, _ := crypto.GenerateEd25519Key(cryptorand.Reader)
-	signingKey, _, _ := crypto.GenerateEd25519Key(cryptorand.Reader)
+	privKey, _, _ := crypto.GenerateEd25519Key(crand.Reader)
+	signingKey, _, _ := crypto.GenerateEd25519Key(crand.Reader)
```
Comment (adlerjohn) [inline rpc/client/client_test.go:93]: This change doesn't seem related to this PR, can you make a new PR for it?

File: rpc/client/client_test.go [hunk 3]
```diff
@@ -537,6 +536,63 @@ func TestUnconfirmedTxsLimit(t *testing.T) {
 	assert.NotContains(txRes.Txs, tx2)
 }
 
+func TestEventIndexing(t *testing.T) {
+	assert := assert.New(t)
+	require := require.New(t)
+
+	mockApp := &mocks.Application{}
+	mockApp.On("InitChain", mock.Anything).Return(abci.ResponseInitChain{})
```
Comment (adlerjohn) [inline rpc/client/client_test.go:544]: Use a constant rather than the literal `"InitChain"` and the rest of the literals in this test. I imagine this is an issue with the rest of the tests too, so maybe file a good first issue to track improving this.

Comment (tzdybal) [reply rpc/client/client_test.go:544]: Issue #349 created.

**Comment by tzdybal** [comment]
After re-thinking this, actually there is no reason to merge this.
